src/main/java/com/bookstore/service/order/OrderServiceImpl.java

```diff
@@ -0,0 +1,119 @@
+package com.bookstore.service.order;
+
+import com.bookstore.dto.order.CreateOrderRequestDto;
+import com.bookstore.dto.order.OrderResponseDto;
+import com.bookstore.dto.order.UpdateStatusRequestDto;
+import com.bookstore.dto.orderitem.OrderItemResponseDto;
+import com.bookstore.exception.EntityNotFoundException;
+import com.bookstore.mapper.OrderItemMapper;
+import com.bookstore.mapper.OrderMapper;
+import com.bookstore.model.Order;
+import com.bookstore.model.OrderItem;
+import com.bookstore.model.ShoppingCart;
+import com.bookstore.model.User;
+import com.bookstore.repository.OrderItemRepository;
+import com.bookstore.repository.OrderRepository;
+import com.bookstore.service.cart.ShoppingCartService;
+import java.math.BigDecimal;
+import java.time.LocalDateTime;
+import java.util.Collection;
+import java.util.List;
+import java.util.Objects;
+import java.util.Set;
+import java.util.stream.Collectors;
+import lombok.RequiredArgsConstructor;
+import org.springframework.data.domain.Pageable;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class OrderServiceImpl implements OrderService {
+    private final OrderRepository orderRepository;
+    private final OrderItemRepository orderItemRepository;
+    private final ShoppingCartService shoppingCartService;
+    private final OrderMapper orderMapper;
+    private final OrderItemMapper orderItemMapper;
+
+    @Override
+    public List<OrderResponseDto> findAll(Pageable pageable) {
+        User user = (User) SecurityContextHolder.getContext().getAuthentication().getPrincipal();
+        Set<OrderItemResponseDto> orderItems =
+                orderRepository.findAllByUserId(user.getId()).stream()
+                .map(Order::getOrderItems)
+                .flatMap(Collection::stream)
+                .map(orderItemMapper::toDto)
+                .collect(Collectors.toSet());
+        return orderRepository.findAllByUserId(user.getId()).stream()
+                .map(orderMapper::toDto)
+                .peek(c -> c.setOrderItems(orderItems))
+                .toList();
```

Comment:
Do you really need all this code? You can do it like findAllOrders() -> stream ()-> toDto -> toList()